az-core/src/main/java/azkaban/Constants.java

```diff
@@ -129,6 +129,11 @@
   // Used (or should be used) wherever a string representation of UTF_8 charset is needed:
   public static final String UTF_8 = java.nio.charset.StandardCharsets.UTF_8.toString();
 
+  // Specifies the source(adhoc, scheduled, flow_trigger) from where flow execution is triggered
+  public static final String EXECUTION_SOURCE_ADHOC = "adhoc";
+  public static final String EXECUTION_SOURCE_SCHEDULED = "scheduled";
+  public static final String EXECUTION_SOURCE_FLOW_TRIGGER = "flow_trigger";
```

Comment:
Shouldn't "source" be a noun? How about?
-USER
-SCHEDULE
-EVENT (because it's an event based trigger mechanism)

Reply:
The earlier names were for execution_type. Makes sense, updated the source names. Since the execution can be triggered from command line which might not be from the user manually, keeping the name as adhoc.